
#Parameters for the vector_field_control package



# Feedback linearization ----------  ----------  ----------

# Distance the control point is moved forward from the robots center
d_feedback: 0.25



# Curve's parameters ----------  ----------  ----------

#alpha = (((x-cx)/a)**(gamma) + ((y-cy)/b)**(gamma))**(1.0/gamma) - 1

# Size and shape parameters
a: 4.0 #stretching on x axis
b: 3.0 #Stretching on y axis
cx: 6.0 #center x
cy: 6.0 #center y
gamma: 4 #shape of curve (should be an even integer)

D_alpha: 1.0 #distance for the control points in the creation of the alpha function

# Parameter to invert the direction that the curve is being circulated
invert_direction: False


# Vector field ----------  ----------  ----------

#Flag to invert the motion of the robot (move backwards)
invert_motion_flag: False

#Name of the topic in which the pose will be obtained
pose_topic_name: /turtle1/pose
# pose_topic_name: /ekf_odom_2
#Type of the topic in which the pose will be obtained (options below)
# PoseTurtle
# TFMessage
# Pose
# Odometry
pose_topic_type: PoseTurtle
# pose_topic_type: Odometry

#Name of the topic in which the forward and angular velocities will be published
cmd_vel_topic_name: /turtle1/cmd_vel
# cmd_vel_topic_name: /cmd_vel

#Name of the topic in which the forward and angular velocities will be published
path_topic_name: /ref_traj


# Convergence gain of the vector field
K_F: 3.3
# Reference forward speed for the robot
vd: 0.7
